src/Stratis.Bitcoin/P2P/PeerAddressManager.cs

```diff
@@ -94,23 +94,28 @@ public interface IPeerAddressManager : IDisposable
     /// </summary>
     public sealed class PeerAddressManager : IPeerAddressManager
     {
+        /// <inheritdoc />
+        public ConcurrentDictionary<IPEndPoint, PeerAddress> Peers { get; private set; }
+
+        /// <summary>The file name of the peers file.</summary>
+        internal const string PeerFileName = "peers.json";
+
+        /// <inheritdoc />
+        public DataFolder PeerFilePath { get; set; }
+
+        /// <summary>Constructor used by unit tests.</summary>
```

Comment:
in general such constructors seem like hacks, is there a plan to remove them in future?

Reply:
I normally don't see an issue with having code that facilitates unit testing... But if you insist I can remove this in this instance. I don't mind either way.

Reply:
i'm not sure, i have a feeling that if you need special constructors for that, something may be wrong with how the thing is designed, but i haven't put a deeper thought into it

Reply:
It is wrong yes, if this is temporary (like I did in node where we did not have an interface) then ok.
But normally we need to inject the dependencies and use mock interfaces.

Reply:
OK guys, I'll remove this constructor 👍

Reply:
And the ones in the subclasses.